src/DonorProfiles/Profile.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+namespace Give\DonorProfiles;
+
+use \Give_Donor as DonorModel;
+
+class Profile {
+
+	protected $donor;
+	protected $id;
+
+	public function __construct( int $donorId ) {
+		$this->id    = $donorId;
+		$this->donor = new DonorModel( $donorId );
+	}
+
+	/**
+	 * Handles updating relevant profile fields in donor database and meta database
+	 *
+	 * @param object $data Object representing profile data to update
+	 *
+	 * @return array
+	 *
+	 * @since 2.11.0
+	 */
+	public function update( $data ) {
+
+		// Handle updates to the donor meta table
+		$this->updateDonorMetaDB( $data );
```

Comment:
Since a number of things are happening here in turn, I would suggest refactoring this out using a "pipeline" pattern, which accepts multiple process classes (each handling a specific part of the update, thus a single responsibility ).

For example:
```php
( new ProfileUpdatePipeline( $data ) )
    ->update( new UpdateDonorMeta )
    ->update( new UpdateAdditionalEmails )
    ->update( new UpdateDonor )
    ->update( new UpdateDonorAddressMeta )
;
```

Reply:
This would also simplify the responsibility of the Profile class, which can focus on formatting and mapping Donor data.

Reply:
Great thinking, really cleared up the Profile class! Done, ref: https://github.com/impress-org/givewp/pull/5486/commits/d1793daf8e0073de0a31fe3f989d7d04eef1f429